### DISCUSSION TITLE
Metagraph error while performing compression

### DISCUSSION THREAD

**Discussion by sinansar123** [tensorflow/compression#136] [General]
When I try to run the pre-trained models for compression I get an error regarding the metagraphs  for the models.
Are some of the models in the repository (such as bls2017) not available anymore? They are not listed in the models list (displayed by tfci.py) either at the moment. 

If possible could you please update the url for those models as well, I would like to experiment with some of them for my bachelor thesis. Thank you

**Comment by jonaballe** [comment]
Hello sinansar123:

The BLS2017 model was never in the list of pre-trained models.

That is because the original implementation in Theano had some properties that aren't reproduced in the TensorFlow implementation in `models/`. For example, it used valid padding throughout, with a large initial reflective padding in the image space. It also used a piecewise linear density model. It would take additional time to reproduce this exactly in TF, and we didn't want to put a pre-trained model there that doesn't reproduce the results, because it could be misleading. The `models/bls2017.py` implementation is just an approximation, as stated in the code.

If you are not trying to exactly reproduce the results from the BLS2017 paper, you could take a look at either the `b2018` or `bmshj2018-factorized` models. They are very similar, but have somewhat different performance due to different architectures and some implementation details as discussed above.

Hope this helps!
Johannes
